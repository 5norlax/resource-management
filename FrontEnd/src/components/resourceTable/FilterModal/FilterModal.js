import React, { useState, useEffect } from 'react'
import axios from 'axios'
import Select from 'react-select'
import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Row, FormGroup } from 'reactstrap'

export default React.memo(function FilterModal({ toggleSkillMatch, hideSkillMatch, notHidden, neededSkills, setNeededSkill }) {

  const [ availSkills, setAvailSkills ] = useState()
  const [ skillFilter, setSkillFilter ] = useState([ ...neededSkills ])
  const [ selectedSkills, setSelectedSkills ] = useState(neededSkills.map(s => ({ label: s, value: s })))

  useEffect(() => {
    axios.get('http://localhost:5000/api/skills')
      .then(res => {
        const resourceData = res.data.Skills.map(s => ({ label: s, value: s }))
        setAvailSkills(resourceData)
      }).catch(error => {
        console.log(error)
      })
  }, [])
  const handleChange = selectedOption => {
    if (selectedOption !== undefined && selectedOption !== null) {
      let skills = selectedOption.map(s => s.value)
      setSkillFilter([ ...skills ])
      setSelectedSkills(selectedOption)
    } else {
      setSkillFilter([])
      setSelectedSkills([])
    }
  }
  const handleFilter = (e) => {
    e.preventDefault()
    if (skillFilter.length === 0) {
      setNeededSkill(skillFilter)
      hideSkillMatch()
      toggle()
    }
    else if (notHidden === true) {
      setNeededSkill(skillFilter)
      toggle()
    }
    else {
      setNeededSkill(skillFilter)
      toggleSkillMatch()
      toggle()
    }
  }
  const handleCancel = (e) => {
    e.preventDefault()
    if (notHidden === true) {
      hideSkillMatch()
      setNeededSkill([])
      setSkillFilter([])
      setSelectedSkills([])
      toggle()
    } else {
      setNeededSkill([])
      setSkillFilter([])
      setSelectedSkills([])
      toggle()
    }
  }
  const [ modal, setModal ] = useState(false)
  const toggle = () => setModal(!modal)

  return (
    <div>
      <Button style={{ height: '50px', textDecoration: 'none', marginTop: '5px', marginBottom: '5px' }} onClick={toggle} className="btn-block shadow-none" type="button"><i className="fas fa-filter"></i>&nbsp;&nbsp;Filter</Button>
      <Modal isOpen={modal} toggle={toggle} autoFocus={false}>
        <ModalHeader className="modalheader" toggle={toggle}>Filter Based on Skills</ModalHeader>
        <ModalBody className="modalbody">
          <Row>
            <FormGroup className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12">
              <label className="sr-only">Skills:</label>
              <Select style={{ backgroundColor: '#fff' }}  autoFocus={true} id="select-search" isMulti options={availSkills} onChange={handleChange} value={selectedSkills} />
            </FormGroup>
          </Row>
        </ModalBody>
        <ModalFooter>
          <Button color="primary" className="blue-button shadow-none" onClick={handleFilter}>Filter</Button>{' '}
          <Button color="secondary" className="shadow-none" onClick={handleCancel}>Remove Filter</Button>
        </ModalFooter>
      </Modal>
    </div>
  )
})