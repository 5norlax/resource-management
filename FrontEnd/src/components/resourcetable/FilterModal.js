import React, { useState } from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Row, Input } from 'reactstrap';
import NeededSkill from './NeededSkill';

const FilterModal = ({ toggleSkillMatch, hideSkillMatch, notHidden, neededSkills, setNeededSkill }) => {

    const [skill, setSkill] = useState('')
    const [skillFilter, setSkillFilter] = useState([...neededSkills])

    const handleChange = (e) => {
        setSkill(e.target.value)
    }
    // Added an event handler for hitting enter to add a new skill to the filter.
    const handleKeyPress = e => {
        if (e.key === "Enter") handleAddSkill(e)
    }
    const handleAddSkill = (e) => {
        e.preventDefault()
        if (skill === '') return
        else {
            setSkillFilter([...skillFilter, skill])
            setSkill('')
        }
    }
    const handleRemoveSkill = (skill) => {
        let remove = skillFilter.filter(s => s !== skill)
        setSkillFilter(remove)
    }
    const handleFilter = (e) => {
        e.preventDefault()
        if (skillFilter.length === 0) {
            setNeededSkill(skillFilter)
            hideSkillMatch()
            toggle()
        }
        else if (notHidden === true) {
            setNeededSkill(skillFilter)
            toggle()
        }
        else {
            setNeededSkill(skillFilter)
            toggleSkillMatch()
            toggle()
        }
    }
    const handleCancel = (e) => {
        e.preventDefault()
        if (notHidden === true) {
            hideSkillMatch()
            setNeededSkill([])
            setSkillFilter([])
            toggle()
        } else {
            setNeededSkill([])
            setSkillFilter([])
            toggle()
        }
    }
    const [modal, setModal] = useState(false);
    const toggle = () => setModal(!modal);

    return (
        <div>
            <Button style={{ height: '50px', textDecoration: 'none' }} onClick={toggle} className="btn-block shadow-none" type="button"><i className="fas fa-filter"></i>&nbsp;&nbsp;Filter</Button>
            <Modal isOpen={modal} toggle={toggle}>
                <ModalHeader className="modalheader" toggle={toggle}>Filter Based on Skills</ModalHeader>
                <ModalBody className="modalbody">
                    <Row>
                        <div className="rounded-input col-xl-9 col-lg-9 col-md-9 col-sm-9 col-xs-9">
                            <div className="p-1 bg-light rounded rounded-pill shadow-sm mb-4">
                                <div className="input-group">
                                    <div className="input-group-prepend">
                                        <button id="button-addon2" type="submit" className="btn btn-link text-primary"><i style={{color: '#007bff'}} className="fa fa-search"></i></button>
                                    </div>
                                    <Input onChange={handleChange} onKeyPress={handleKeyPress} style={{ marginRight: '15px', marginLeft: '15px' }} value={skill} type="search" id="myInput" placeholder="Search for Skill" aria-describedby="button-addon2" className="form-control border-0 bg-light" />
                                </div>
                            </div>
                        </div>
                        <div className="col-xl-3 col-lg-3 col-md-3 col-sm-3 col-xs-3">
                            <Button onClick={handleAddSkill} id="add-skill" className="shadow-none" style={{ marginTop: '5px' }}>
                                <i style={{color: 'inherit'}} className="fas fa-plus-circle fa-lg"></i>&nbsp;&nbsp;Add</Button>
                        </div>
                    </Row>
                    <NeededSkill skills={skillFilter} removeSkill={handleRemoveSkill} />
                </ModalBody>
                <ModalFooter>
                    <Button color="primary" className="shadow-none" onClick={handleFilter}>Filter</Button>{' '}
                    <Button color="secondary" className="shadow-none" onClick={handleCancel}>Remove Filter</Button>
                </ModalFooter>
            </Modal>
        </div>
    );
}

export default FilterModal;