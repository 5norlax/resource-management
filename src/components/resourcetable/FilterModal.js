import React, { useState } from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Row, Input } from 'reactstrap';
import NeededSkill from './NeededSkill';

const FilterModal = ({ toggleSkillMatch, hideSkillMatch, notHidden, neededSkills, setNeededSkill, filterTable }) => {
    
    const [skill, setSkill] = useState('')
    
    const handleChange = (e) => {
        setSkill(e.target.value)
    }
    const handleAddSkill = (e) => {
        e.preventDefault()
        if (skill === '') return
        else { 
        setNeededSkill([...neededSkills, skill])
        setSkill('') }
    }
    const handleRemoveSkill = (skill) => {
        let remove = neededSkills.filter(s => s !== skill)
        setNeededSkill(remove)
    }
    const handleFilter = (e) => {
        e.preventDefault()
        if (neededSkills.length === 0) {
            hideSkillMatch()
            toggle()
        }
        else if (notHidden === true) {
            toggle()
        }
        else {
            toggleSkillMatch()
            toggle()
        }
    }
    const handleCancel = (e) => {
        e.preventDefault()
        if (notHidden === true) {
            hideSkillMatch()
            setNeededSkill([])
            toggle() 
        } else {
            setNeededSkill([])
            toggle()
        }
    }
    const [modal, setModal] = useState(false);
    const toggle = () => setModal(!modal);

    return (
        <div>
            <Button style={{ height: '50px' }} onClick={toggle} className="shadow-none" type="button"><i className="fas fa-filter"></i>&nbsp;&nbsp;Filter</Button>
            <Modal isOpen={modal} toggle={toggle}>
                <ModalHeader toggle={toggle}>Filter Based on Skills</ModalHeader>
                <ModalBody>
                    <Row>
                        <div className="col-xl-9 col-lg-9 col-md-9 col-sm-9 col-xs-9">
                            <div className="p-1 bg-light rounded rounded-pill shadow-sm mb-4" style={{}}>
                                <div className="input-group">
                                    <div className="input-group-prepend">
                                        <button id="button-addon2" type="submit" className="btn btn-link text-primary"><i className="fa fa-search"></i></button>
                                    </div>
                                    <Input onChange={handleChange} value={skill} type="search" id="myInput" placeholder="Search for Skill" aria-describedby="button-addon2" className="form-control border-0 bg-light" />
                                </div>
                            </div>
                        </div>
                        <div className="col-xl-3 col-lg-3 col-md-3 col-sm-3 col-xs-3">
                            <Button onClick={handleAddSkill} className="shadow-none" style={{ color: '#007bff', backgroundColor: 'white', borderColor: 'white', marginTop: '5px'}}><i className="fas fa-plus-circle fa-lg"></i> Add</Button>
                        </div>
                    </Row>
                    <NeededSkill skills={neededSkills} removeSkill={handleRemoveSkill} />
                </ModalBody>
                <ModalFooter>
                    <Button color="primary" className="shadow-none" onClick={handleFilter}>Filter</Button>{' '}
                    <Button color="secondary" className="shadow-none" onClick={handleCancel}>Remove Filter</Button>
                </ModalFooter>
            </Modal>
        </div>
    );
}

export default FilterModal;