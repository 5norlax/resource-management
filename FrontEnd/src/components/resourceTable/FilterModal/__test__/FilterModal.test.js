import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import FilterModal from '../FilterModal';

afterEach(cleanup);

describe('Test FilterModal', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<FilterModal neededSkills={['Java']}/>).toJSON()
    expect(tree).toMatchSnapshot();
  });
  
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FilterModal neededSkills={['Java']} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

});