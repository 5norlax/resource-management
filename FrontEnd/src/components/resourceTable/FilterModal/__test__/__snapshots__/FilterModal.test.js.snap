// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FilterModal matches snapshot 1`] = `
<div>
  <button
    aria-label={null}
    className="btn-block shadow-none btn btn-secondary"
    onClick={[Function]}
    style={
      Object {
        "height": "50px",
        "marginBottom": "5px",
        "marginTop": "5px",
        "textDecoration": "none",
      }
    }
    type="button"
  >
    <i
      className="fas fa-filter"
    />
      Filter
  </button>
</div>
`;
